File: backend/internal/model/vo/scan_job.go
package vo

import "antd-gin-admin-backend/internal/model/entity"

// ScanJobVO is the API view of a Scan Job.
type ScanJobVO struct {
	JobCode      string  `json:"job_code"`
	AssetCode    string  `json:"asset_code,omitempty"`
	EntryURL     string  `json:"entry_url"`
	DeptCode     string  `json:"dept_code"`
	Policy       string  `json:"policy"`
	MaxDepth     int     `json:"max_depth"`
	MaxPages     int     `json:"max_pages"`
	Status       string  `json:"status"`
	FindingCount int     `json:"finding_count"`
	TempURL      bool    `json:"temp_url"`
	CreatedAt    string  `json:"created_at"`
	FinishedAt   *string `json:"finished_at,omitempty"`
}

func BuildScanJobVO(j *entity.ScanJob) *ScanJobVO {
	if j == nil {
		return nil
	}
	vo := &ScanJobVO{
		JobCode:      j.JobCode,
		AssetCode:    j.AssetCode,
		EntryURL:     j.EntryURL,
		DeptCode:     j.DeptCode,
		Policy:       j.Policy,
		MaxDepth:     j.MaxDepth,
		MaxPages:     j.MaxPages,
		Status:       j.Status,
		FindingCount: j.FindingCount,
		TempURL:      j.TempURL,
		CreatedAt:    j.CreatedAt.Format("2006-01-02 15:04:05"),
	}
	if j.FinishedAt != nil {
		s := j.FinishedAt.Format("2006-01-02 15:04:05")
		vo.FinishedAt = &s
	}
	return vo
}

File: frontend/admin-web/src/services/antd-gin-api/scan_job.ts
import { request } from '@umijs/max';
import type { PageResult } from './asset';

export interface ScanJob {
  job_code: string;
  asset_code?: string;
  entry_url: string;
  dept_code: string;
  policy: 'quick' | 'standard' | 'deep';
  max_depth: number;
  max_pages: number;
  status: 'queued' | 'running' | 'succeeded' | 'failed' | 'cancelled';
  finding_count: number;
  temp_url: boolean;
  created_at: string;
  finished_at?: string;
}

export interface CreateScanJobRequest {
  source: 'asset' | 'temp_url';
  asset_code?: string;
  entry_url?: string;
  policy: 'quick' | 'standard' | 'deep';
  max_depth?: number;
  max_pages?: number;
}

export interface ListScanJobParams {
  page?: number;
  pageSize?: number;
  status?: string;
  policy?: string;
  keyword?: string;
}

/** 获取扫描任务列表 */
export async function getScanJobList(params?: ListScanJobParams) {
  return request<{
    code: number;
    data: PageResult<ScanJob>;
    message?: string;
  }>('/api/v1/scan/scan-jobs', {
    method: 'GET',
    params,
  });
}

/** 获取扫描任务详情 */
export async function getScanJob(jobCode: string) {
  return request<{
    code: number;
    data: ScanJob;
    message?: string;
  }>(`/api/v1/scan/scan-jobs/${jobCode}`, {
    method: 'GET',
  });
}

/** 创建扫描任务 */
export async function createScanJob(data: CreateScanJobRequest) {
  return request<{
    code: number;
    data: ScanJob;
    message?: string;
  }>('/api/v1/scan/scan-jobs', {
    method: 'POST',
    data,
  });
}

/** 取消扫描任务 */
export async function cancelScanJob(jobCode: string) {
  return request<{
    code: number;
    data: ScanJob;
    message?: string;
  }>(`/api/v1/scan/scan-jobs/${jobCode}/cancel`, {
    method: 'POST',
  });
}
